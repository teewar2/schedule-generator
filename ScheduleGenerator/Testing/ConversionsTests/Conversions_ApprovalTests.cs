using System.Collections.Generic;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using Domain.Conversions;
using Domain.MeetingsParts;
using Infrastructure.SheetPatterns;
using NUnit.Framework;
using static Infrastructure.SheetConstants;


namespace Testing.ConversionsTests
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class Conversions_ApprovalTests
    {
        //TODO make actual test
        [Test]
        public void TimeRequisitionsReadTest()
        {
            var requisitionData = SheetTableReader
                .ReadRowsFromSheet(Repository, InputRequirementsTestSheetName, 1, 0, 7)
                .Take(10);
            //var allTimeRequisitions = new List<List<MeetingTimeRequisition>>();
            var checkStrings = new List<string>();
            foreach (var requisitionRow in requisitionData)
            {
                var meetingTimesRaw = requisitionRow[5];
                var meetingTimeRequisitions =
                    SheetToRequisitionConverter.ParseMeetingTimeRequisitions(meetingTimesRaw);
                var a = meetingTimeRequisitions.Select(r => string.Join('\n',r.MeetingTimeChoices.Select(m=>m.ToString())));
                var checkString = 
                    $"{meetingTimesRaw}\n{string.Join("\n\n", a)}";
                checkStrings.Add(checkString);
            }
            var unifiedString = string.Join("\n------\n", checkStrings);
            Approvals.Verify(unifiedString);
        }
    }
}