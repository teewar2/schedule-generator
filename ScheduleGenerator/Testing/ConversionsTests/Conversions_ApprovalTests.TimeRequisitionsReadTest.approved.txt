пн: 1-3 пара 
пт: 4-6 пара
Day: Monday, TimeSlotIndex: 1
Day: Monday, TimeSlotIndex: 2
Day: Monday, TimeSlotIndex: 3

Day: Friday, TimeSlotIndex: 4
Day: Friday, TimeSlotIndex: 5
Day: Friday, TimeSlotIndex: 6
------
пн, ср, пт-сб: 2-5 пара
Day: Monday, TimeSlotIndex: 2
Day: Monday, TimeSlotIndex: 3
Day: Monday, TimeSlotIndex: 4
Day: Monday, TimeSlotIndex: 5
Day: Wednesday, TimeSlotIndex: 2
Day: Wednesday, TimeSlotIndex: 3
Day: Wednesday, TimeSlotIndex: 4
Day: Wednesday, TimeSlotIndex: 5
Day: Friday, TimeSlotIndex: 2
Day: Friday, TimeSlotIndex: 3
Day: Friday, TimeSlotIndex: 4
Day: Friday, TimeSlotIndex: 5
Day: Saturday, TimeSlotIndex: 2
Day: Saturday, TimeSlotIndex: 3
Day: Saturday, TimeSlotIndex: 4
Day: Saturday, TimeSlotIndex: 5
------
вт,ср,пт: 1-4 пара
Day: Tuesday, TimeSlotIndex: 1
Day: Tuesday, TimeSlotIndex: 2
Day: Tuesday, TimeSlotIndex: 3
Day: Tuesday, TimeSlotIndex: 4
Day: Wednesday, TimeSlotIndex: 1
Day: Wednesday, TimeSlotIndex: 2
Day: Wednesday, TimeSlotIndex: 3
Day: Wednesday, TimeSlotIndex: 4
Day: Friday, TimeSlotIndex: 1
Day: Friday, TimeSlotIndex: 2
Day: Friday, TimeSlotIndex: 3
Day: Friday, TimeSlotIndex: 4
------
пн-пт: 3,4 пара
Day: Monday, TimeSlotIndex: 3
Day: Monday, TimeSlotIndex: 4
Day: Tuesday, TimeSlotIndex: 3
Day: Tuesday, TimeSlotIndex: 4
Day: Wednesday, TimeSlotIndex: 3
Day: Wednesday, TimeSlotIndex: 4
Day: Thursday, TimeSlotIndex: 3
Day: Thursday, TimeSlotIndex: 4
Day: Friday, TimeSlotIndex: 3
Day: Friday, TimeSlotIndex: 4
------
пн, ср, пт: 2-5 пара
Day: Monday, TimeSlotIndex: 2
Day: Monday, TimeSlotIndex: 3
Day: Monday, TimeSlotIndex: 4
Day: Monday, TimeSlotIndex: 5
Day: Wednesday, TimeSlotIndex: 2
Day: Wednesday, TimeSlotIndex: 3
Day: Wednesday, TimeSlotIndex: 4
Day: Wednesday, TimeSlotIndex: 5
Day: Friday, TimeSlotIndex: 2
Day: Friday, TimeSlotIndex: 3
Day: Friday, TimeSlotIndex: 4
Day: Friday, TimeSlotIndex: 5
------
пн,пт: 3-4 пара; вт:2 пара
Day: Monday, TimeSlotIndex: 3
Day: Monday, TimeSlotIndex: 4
Day: Friday, TimeSlotIndex: 3
Day: Friday, TimeSlotIndex: 4
Day: Tuesday, TimeSlotIndex: 2
------
пн: 1-3 пара 
пт: 4,6 пара
Day: Monday, TimeSlotIndex: 1
Day: Monday, TimeSlotIndex: 2
Day: Monday, TimeSlotIndex: 3

Day: Friday, TimeSlotIndex: 4
Day: Friday, TimeSlotIndex: 6
------

Day: Monday, TimeSlotIndex: 1
Day: Monday, TimeSlotIndex: 2
Day: Monday, TimeSlotIndex: 3
Day: Monday, TimeSlotIndex: 4
Day: Monday, TimeSlotIndex: 5
Day: Monday, TimeSlotIndex: 6
Day: Tuesday, TimeSlotIndex: 1
Day: Tuesday, TimeSlotIndex: 2
Day: Tuesday, TimeSlotIndex: 3
Day: Tuesday, TimeSlotIndex: 4
Day: Tuesday, TimeSlotIndex: 5
Day: Tuesday, TimeSlotIndex: 6
Day: Wednesday, TimeSlotIndex: 1
Day: Wednesday, TimeSlotIndex: 2
Day: Wednesday, TimeSlotIndex: 3
Day: Wednesday, TimeSlotIndex: 4
Day: Wednesday, TimeSlotIndex: 5
Day: Wednesday, TimeSlotIndex: 6
Day: Thursday, TimeSlotIndex: 1
Day: Thursday, TimeSlotIndex: 2
Day: Thursday, TimeSlotIndex: 3
Day: Thursday, TimeSlotIndex: 4
Day: Thursday, TimeSlotIndex: 5
Day: Thursday, TimeSlotIndex: 6
Day: Friday, TimeSlotIndex: 1
Day: Friday, TimeSlotIndex: 2
Day: Friday, TimeSlotIndex: 3
Day: Friday, TimeSlotIndex: 4
Day: Friday, TimeSlotIndex: 5
Day: Friday, TimeSlotIndex: 6
Day: Saturday, TimeSlotIndex: 1
Day: Saturday, TimeSlotIndex: 2
Day: Saturday, TimeSlotIndex: 3
Day: Saturday, TimeSlotIndex: 4
Day: Saturday, TimeSlotIndex: 5
Day: Saturday, TimeSlotIndex: 6
------
вт: 1-5 пара
пт: 3-6 пара
Day: Tuesday, TimeSlotIndex: 1
Day: Tuesday, TimeSlotIndex: 2
Day: Tuesday, TimeSlotIndex: 3
Day: Tuesday, TimeSlotIndex: 4
Day: Tuesday, TimeSlotIndex: 5

Day: Friday, TimeSlotIndex: 3
Day: Friday, TimeSlotIndex: 4
Day: Friday, TimeSlotIndex: 5
Day: Friday, TimeSlotIndex: 6
------

Day: Monday, TimeSlotIndex: 1
Day: Monday, TimeSlotIndex: 2
Day: Monday, TimeSlotIndex: 3
Day: Monday, TimeSlotIndex: 4
Day: Monday, TimeSlotIndex: 5
Day: Monday, TimeSlotIndex: 6
Day: Tuesday, TimeSlotIndex: 1
Day: Tuesday, TimeSlotIndex: 2
Day: Tuesday, TimeSlotIndex: 3
Day: Tuesday, TimeSlotIndex: 4
Day: Tuesday, TimeSlotIndex: 5
Day: Tuesday, TimeSlotIndex: 6
Day: Wednesday, TimeSlotIndex: 1
Day: Wednesday, TimeSlotIndex: 2
Day: Wednesday, TimeSlotIndex: 3
Day: Wednesday, TimeSlotIndex: 4
Day: Wednesday, TimeSlotIndex: 5
Day: Wednesday, TimeSlotIndex: 6
Day: Thursday, TimeSlotIndex: 1
Day: Thursday, TimeSlotIndex: 2
Day: Thursday, TimeSlotIndex: 3
Day: Thursday, TimeSlotIndex: 4
Day: Thursday, TimeSlotIndex: 5
Day: Thursday, TimeSlotIndex: 6
Day: Friday, TimeSlotIndex: 1
Day: Friday, TimeSlotIndex: 2
Day: Friday, TimeSlotIndex: 3
Day: Friday, TimeSlotIndex: 4
Day: Friday, TimeSlotIndex: 5
Day: Friday, TimeSlotIndex: 6
Day: Saturday, TimeSlotIndex: 1
Day: Saturday, TimeSlotIndex: 2
Day: Saturday, TimeSlotIndex: 3
Day: Saturday, TimeSlotIndex: 4
Day: Saturday, TimeSlotIndex: 5
Day: Saturday, TimeSlotIndex: 6